yt_dlp/extractor/bilibili.py

```diff
@@ -938,20 +940,44 @@ def _get_subtitles(self, *, ep_id=None, aid=None):
                 'episode_id': ep_id,
                 'aid': aid,
             })) or {}
-        subtitles = {}
-        for sub in sub_json.get('subtitles') or []:
-            sub_url = sub.get('url')
-            if not sub_url:
-                continue
-            sub_data = self._download_json(
-                sub_url, ep_id or aid, errnote='Unable to download subtitles', fatal=False,
-                note='Downloading subtitles%s' % f' for {sub["lang"]}' if sub.get('lang') else '')
-            if not sub_data:
-                continue
-            subtitles.setdefault(sub.get('lang_key', 'en'), []).append({
-                'ext': 'srt',
-                'data': self.json2srt(sub_data)
-            })
+
+        for sub in traverse_obj(sub_json, ('subtitles', lambda _, v: v['url'])):
+            sub_ext = determine_ext(sub['url'])
+            sub_lang = sub.get('lang_key') or 'en'
+
+            sub_data = self._download_webpage(
+                sub['url'], ep_id or aid, fatal=False,
+                encoding='utf-8-sig' if sub_ext == 'ass' else None,
```

Comment:
Specifying the encoding like this is not necessary for `_download_webpage`.
```suggestion
```
If it's not specified/specifying `utf-8`, the BOM will be part of the returned string. While with `utg-8-sig` the BOM will be removed. Having the BOM removed from the string would cause the file to be written without BOM.
This is not what we want, since ASS files should generally be written with CRLF and BOM.

Reply:
A utf8 bom is almost always useless though. Imo the bom should always be stripped.

Reply:
Saving and distributing ASS files with a BOM is still recommended, since a lot of the legacy tooling surrounding ASS files expects and sometimes requires a BOM. The most commonly used editor for ASS files, Aegisub, still sometimes trips up on files that don't have a BOM due to misidentifying the encoding.

imo. Bilibili serves the file with a BOM, so it should be saved with a BOM.

Reply:
On the contrary Ive had problems with files that HAD a BOM, and generally using a BOM seems to be deprecated more and more...

Reply:
Okay, never mind, it's better to strip the BOM.